synapse/federation/sender/per_destination_queue.py

```diff
@@ -430,6 +573,21 @@ async def _get_device_update_edus(self, limit: int) -> Tuple[List[Edu], int]:
 
         return (edus, now_stream_id)
 
+    def reset_catch_up_state(self) -> None:
+        """
+        Resets the catch-up state of this destination.
+        This does the following:
+         - marks catch-up mode
+         - unsets the catch-up limit (max. stream order)
+             so that it is reset to the highest catch-up next time we have a
+             chance to catch up
+         - empties the main PDU queue as any PDUs in it will now be handled
+             by catch-up (because the max stream order limit was unset).
+        """
+        self._pending_pdus = []
+        self._catching_up = True
+        self._catch_up_max_stream_order = None
+
```

Comment:
This was obviously written before #7864 and so on, I think it's irrelevant, will deal with on monday